### ISSUE TITLE
composer keeps on saying "You must set up the project dependencies using `composer install` ..."

### ISSUE THREAD

**Issue by KelvinNjihia** [composer/composer#7499]
I am trying to install composer on Antergos using the official guide. 

It has completed without errors without errors.

But whenever `composer` or `composer -V`, I keep getting this output.

```
You must set up the project dependencies using `composer install`
See https://getcomposer.org/download/ for instructions on installing Composer
```
On going the link and repeating the process or even running `composer install`, It keeps giving the same output. I want to install composer so that I can be able to install Laravel

**Comment by stof** [comment]
how are you installing composer ?

**Comment by KelvinNjihia** [comment]
I am using this official guide provided here. https://getcomposer.org/download/

Just as they are.
![screenshot from 2018-07-27 13-39-12](https://user-images.githubusercontent.com/5685402/43316609-80712a02-91a2-11e8-931c-1442395d461a.png)

**Comment by KelvinNjihia** [comment]
I have found where the issue was, composer isn't working with php version 7.3.0 alpha. Maybe until that issue is resolved.

**Comment by Jasu** [comment]
Does not work in beta either.  As a workaround, you can extract the PHAR:

```bash
php -r '(new Phar("composer.phar"))->extractTo("composer-extracted");' 
```
and run Composer with `php composer-extracted/bin/composer`.

This might be related to <https://bugs.php.net/bug.php?id=76510>.

**Comment by hal869** [comment]
@Jasu 

> This might be related to https://bugs.php.net/bug.php?id=76510.

fwiw, the workaround there, adding "-fno-optimize-strlen" to build flags, does the trick.  with 

```
php -v
	PHP 7.3.0-dev (cli) (built: Aug  7 2018 07:23:47) ( NTS )
	Copyright (c) 1997-2018 The PHP Group
	Zend Engine v3.3.0-dev, Copyright (c) 1998-2018 Zend Technologies
	    with Zend OPcache v7.3.0-dev, Copyright (c) 1999-2018, by Zend Technologies

composer --version
	Composer version 1.7.1 2018-08-07 09:39:23
```

no more "You must set up the project dependencies ..." issue.

Will retest once an official 'fix' is in place.

**Comment by ktomk** [comment]
Does it make sense to have the "You must set up the project dependencies ..."  error message in case composer is packaged as phar? This *could* constitute a bug as the error message is misleading: Here this is an issue with the PHP runtime and not w/ missing dependencies. The phar package should never miss dependencies hence the error message is wrong.

**Comment by videni** [comment]
also have this problem on php7.4.13

**Comment by ktomk** [comment]
@videni: is composer installed as phar (guess so, just clarifaying). and is this problem new? was there a configuration change like switching xdebug version 2 to xdebug version 3 or something similiar?

**Comment by videni** [comment]
@ktomk , thanks for your quick reply,  both composer 1.10.19, composer 2.0.8 has the same error on php7.4.13, no xdebug installed

**Comment by ktomk** [comment]
@videni can you tell where/how you obtained the php binary on the system where it fails? and perhaps some details about the system. from the original report this was related to how the php binary was compiled. I've also tested with PHP 7.4.13 and for me it's all fine:

```
$ php7.4 -n -dextension=phar -dextension=json -f $(which composer) -- -V
Composer version 1.10.17 2020-10-30 22:31:58
```

Using Ubuntu 18.04 and PHP from Sury.org.

**Comment by nathandentzau** [comment]
I ran into this today with a project still on PHP 7.4. Defining `composer_version: 2.2.10` fixed the issue for me.

**Comment by dalin-** [comment]
Ran into this today within a Docker container.  The problem was that composer was also included in the PHP project, and the $PATH points to both the global composer and the project-specific composer, and the first time I ran `composer install` it failed partway, leaving a corrupted project-specific composer.  

The solution was to directly call the global composer to try again.  In my case that was `/usr/local/bin/composer install`.  You can probably find it via `whereis composer`.

**Comment by BaluErtl** [comment]
Ugh, that was a tough one. After messing up the package from one of my locally symlinked path repositories, Composer stuck in the installation process halfway. Any command I tried resulted in the same output:
```
You must set up the project dependencies using composer install
See https://getcomposer.org/download/ for instructions on installing Composer
```
Restarting, and rebuilding Lando containers (or even OrbStack) did not help, still the same.

But thanks to Dalin, your hint of checking `$ whereis composer` revealed that there's a last resort. Calling the global executable within the `appserver` container I was able to gradually build back everything. Some commands I used in various order:

- `$ /usr/local/bin/composer clear-cache`
- `$ /usr/local/bin/composer diagnose`
- `$ /usr/local/bin/composer self-update`
- `$ /usr/local/bin/composer install`
- `$ /usr/local/bin/composer update --lock`
